rules/rules-reviewed/eap8/eap6/eap8-xml-binding.windup.xml

```diff
@@ -0,0 +1,161 @@
+<?xml version="1.0"?>
+<ruleset xmlns="http://windup.jboss.org/schema/jboss-ruleset" id="eap8-xml-binding"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This ruleset provides analysis of XML Binding for migration to JBoss EAP 8.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final" />
+        </dependencies>
+        <sourceTechnology id="eap" versionRange="[6,7]" />
+        <targetTechnology id="eap" versionRange="[8,)" />
+        <tag>eap8</tag>
+    </metadata>
+    <rules>
+        <rule id="eap8-xml-binding-00001">
+            <when>
+                <or>
+                    <javaclass references="com.sun.xml.bind.api.{*}">
+                        <location>IMPORT</location>
+                    </javaclass>
+                    <javaclass references="com.sun.xml.bind.v2.{*}">
+                        <location>IMPORT</location>
+                    </javaclass>
+                </or>
+            </when>
+            <perform>
+                <hint title="Implementation class packages must be updated" effort="1" category-id="mandatory">
+                    <message>Implementation class packages must be updated to Jakarta XML Binding 4.0 for EAP8.
+                        Update package paths  from `com.sun.xml.bind.*` to `org.glassfish.jaxb.runtime.*`</message>
+                    <link title="Jakarta XML Binding 4.0 specification"
+                          href="https://jakarta.ee/specifications/xml-binding/4.0/"/>
+                    <tag>eap8</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="eap8-xml-binding-00002">
+            <when>
+                <javaclass references="javax.xml.bind.Marshaller.setProperty({*})" as="setPropertiesMethodCalls">
```

Comment:
Doesn't this work with just `<javaclass references="javax.xml.bind.Marshaller.setProperty({property})">...` and then a `<where param="property"><matches pattern="com.sun.xml.bind{*}"></where>`?

Reply:
I don't think it would because the method parameters are specified in the rule as a type reference not as  a particular value so you'd have to specify (java.lang,String, java.lang.Object) which doesn't drill down to the value we want.